osu.Game/IO/CopyOnWriteHelper.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace osu.Game.IO
+{
+    internal static class CopyOnWriteHelper
+    {
+        public static bool CheckAvailability(string testDestinationPath, string testSourcePath)
+        {
+            if (OperatingSystem.IsWindowsVersionAtLeast(10, 0, 26100))
+            {
+                // Starting from Windows 11 24H2, the CopyFile syscall will perform CoW cloning on ReFS drives.
```

Comment:
I'm not able to find this information in the [CopyFile docs](https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-copyfile). I'm only seeing that ReFS is supported since Windows 8. Is there another place I should look? (Link it in the code comment!)

Reply:
https://github.com/dotnet/runtime/issues/86681#issuecomment-1780510385

Announcement blogs:
https://blogs.windows.com/windows-insider/2023/10/25/announcing-windows-11-insider-preview-build-25982-canary-channel/
https://devblogs.microsoft.com/engineering-at-microsoft/copy-on-write-in-win32-api-early-access/
https://devblogs.microsoft.com/engineering-at-microsoft/copy-on-write-performance-and-debugging/

Reply:
They are discussions and blog posts instead of formal documentation, so I'm not sure whether it's a good choice to include their links.